### DISCUSSION TITLE
Grand Total Stats

### DISCUSSION THREAD

**Discussion by homeautomaton** [merbanan/rtl_433#2452] [Show and tell]
I love when I start thinking about a possible feature request, then I remember to RTFM (again) only to find that rtl_433 can already do basically what I want. It's a really excellent project. Kudos to the maintainers.

I originally overlooked the "stats" option of the -M feature, thinking **"Add various meta data to each output"** meant the options all applied to the regular reception-based messages so I skipped reading all of the details. Later when I re-read the man page and saw the time setting here **"Use "stats[:[<level>][:<interval>]]" to report statistics (default: 600 seconds),"** I realized that this is _almost_ exactly what I was looking for.

I'm probably not the only user that runs rtl_433 for the purpose of finding other interesting RF transmissions nearby. I'd been using "grep" "sed" and "uniq" to post process the detailed log (as well as storing data in a sqlite db) to see what kinds of signals repeat over a long period of time. (Anything received just once might be noise.) Now I'm using "report_meta stats:1:99999999" in my config, or the equivalent option "-M stats:1:99999999" on the command-line.

It might not be obvious how cool this feature is, and why I have a huge number for the time, but what this does is print the final stats when I stop the program with control-C.

So the feature I originally wanted, a summary of everything that's been received, already exists, in a way. I could still imagine a separate "grand total" feature, so that periodic stats could also be seen. (This presently resets all of the counters, which is why I have opted to use a really long time setting, and periodically stop to see what I've found.)

**Comment by zuckschwerdt** [comment]
We already planned to add that -- sort of ;)
https://github.com/merbanan/rtl_433/blob/3264d01d65f386529619eecadf1471c4cf42ee5a/src/rtl_433.c#L1091

**Comment by homeautomaton** [comment]
I've discovered that if I redirect output to a file `rtl_433 -M stats:1:99999999 2>&1 | tee -a /tmp/rtl.log`, I'm not getting the stats report when I stop with control-C.

Is this a desired behavior?

**Reply by zuckschwerdt** [reply]
Well, by design at least. If you cancel (<kbd>CTRL-C</kbd>, `SIGINT`) a pipe all programs stop. So `tee` is killed too and the output is lost.
Note that a simpler command to show this is `rtl_433 -M stats:1:99999999 2>&1 | cat -` or even `rtl_433 2>&1 | cat -` and then watch "Signal caught, exiting!" disappear.

You should do this (if you really want to log KV output): `rtl_433 -M stats:1:99999999 -F kv:/tmp/rtl.log`
Or `rtl_433 -M stats:1:99999999 -F kv,v=8:/tmp/rtl.log` to capture the log/error messages too (**but this was broken and must be written** `-F kv,v=8/tmp/rtl.log` before 0ce0501)

**Reply by homeautomaton** [reply]
Another aha moment. I didn't relate the "format" type under "-F" called "states" to mean "stats"!

I have everything I need in the MQTT feed, if I enable "states". I'll accumulate totals in a database.

**Reply by zuckschwerdt** [reply]
Ah yes, `states` -- I couldn't find any other good word to convey "everything that is not an RF event" :)
